web/src/message_fetch.js

```diff
@@ -49,6 +51,15 @@ function process_result(data, opts) {
 
     if (messages.length !== 0) {
         message_util.add_old_messages(messages, opts.msg_list);
+
+        // We keep track of how far back we've fetched messages for, for messaging in
+        // the recent view. This assumes `data.messages` is already sorted.
+        const oldest_timestamp = data.messages[messages.length - 1].timestamp;
+        recent_view_ui.set_oldest_message_date(
+            oldest_timestamp,
+            has_found_oldest,
+            has_found_newest,
+        );
```

Comment:
This code path is shared for all message fetches -- including narrowing to a random view.

You probably need this logic to be inside a `if (opts.msg_list === message_lists.home)` conditional.

Reply:
Great thanks, added that

Reply:
Hm, I think this makes it so that if you start a page load in a narrow and switch to recent conversations, the banner won't show up for a while.